kernel/comps/keyboard/src/i8042_chip/mouse.rs

```diff
@@ -0,0 +1,475 @@
+                }
+            )*
+        };
+    }
+
+    define_commands! {
+        GetDeviceId, 0xF2, fn([u8; 0]) -> [u8; 1];
+        SetSampleRate, 0xF3, fn([u8; 1]) -> [u8; 0];
+        EnableDataReporting, 0xF4, fn([u8; 0]) -> [u8; 0];
+    }
+}
+
+#[derive(Debug)]
+struct I8042Mouse {
+    name: String,
+    phys: String,
+    uniq: String,
+    id: InputId,
+    capability: InputCapability,
+}
+
+impl I8042Mouse {
+    fn new() -> Self {
+        let mut capability = InputCapability::new();
+
+        // Mouse supports key events and relative movement events.
+        capability.set_supported_event_type(aster_input::event_type_codes::EventTypes::KEY);
+        capability.set_supported_event_type(aster_input::event_type_codes::EventTypes::REL);
+        capability.set_supported_event_type(aster_input::event_type_codes::EventTypes::SYN);
+
+        // Add mouse buttons.
+        capability.set_supported_key(KeyCode::BtnLeft);
+        capability.set_supported_key(KeyCode::BtnRight);
+        capability.set_supported_key(KeyCode::BtnMiddle);
+
+        // Add relative axes for movement.
+        capability.set_supported_relative_axis(RelCode::X);
+        capability.set_supported_relative_axis(RelCode::Y);
+        capability.set_supported_relative_axis(RelCode::Wheel);
+
+        Self {
+            // Standard name for i8042 PS/2 mouse devices.
+            name: "i8042_mouse".to_string(),
+
+            // Physical path describing the device's connection topology
+            // isa0060: ISA bus port 0x60 (i8042 data port)
+            // serio1: Serial I/O device 1 (second PS/2 port)
+            // input0: Input device 0 (first mouse device on this controller)
+            phys: "isa0060/serio1/input0".to_string(),
+
+            // Unique identifier - empty string because traditional i8042 mice
+            // don't have unique hardware identifiers
+            uniq: "".to_string(),
+
+            // Device ID with standard values for i8042 mice
+            // BUS_I8042 (0x11): PS/2 interface bus type
+            // vendor (0x0002): Generic vendor ID for standard mice
+            // product (0x0001): Generic product ID for standard mice
+            // version (0x0001): Version 1.0 - standard PS/2 mouse protocol
+            id: InputId::new(InputId::BUS_I8042, 0x0002, 0x0001, 0x0001),
+
+            capability,
+        }
+    }
+}
+
+impl InputDevice for I8042Mouse {
+    fn name(&self) -> &str {
+        &self.name
+    }
+
+    fn phys(&self) -> &str {
+        &self.phys
+    }
+
+    fn uniq(&self) -> &str {
+        &self.uniq
+    }
+
+    fn id(&self) -> InputId {
+        self.id
+    }
+
+    fn capability(&self) -> &InputCapability {
+        &self.capability
+    }
+}
+
+fn handle_mouse_input(_trap_frame: &TrapFrame) {
+    let Some(controller) = I8042_CONTROLLER.get() else {
+        return;
+    };
+    let Some(data) = controller.lock().receive_data() else {
+        return;
```

Comment:
```rust
    let Some(data) = controller.lock().receive_data() else {
        log::warn!("PS/2 mouse has no input data");
        return;
    };
```

Having a warning here is slightly more consistent with the keyboard's `ScancodeInfo::read` implementation.